src/ConnectComponents.tsx

```diff
@@ -13,12 +13,33 @@ ConnectComponentsContext.displayName = 'ConnectComponents';
 export const ConnectComponentsProvider = ({
   connectInstance,
   children,
+  initParams,
 }: {
   connectInstance: connectJs.StripeConnectInstance;
   children: any;
+  initParams?: connectJs.IStripeConnectInitParams | undefined;
 }): JSX.Element => {
+  const [storedConnectInstance, setStoredConnectInstance] =
+    React.useState<connectJs.StripeConnectInstance>(connectInstance);
+
+  if (storedConnectInstance === null && !initParams) {
+    throw new Error();
+  }
+
+  React.useEffect(() => {
+    if (connectInstance === null && initParams) {
+      const createInstance = async () => {
+        const connectWrapper = await connectJs.loadConnect();
+        setStoredConnectInstance(await connectWrapper.initialize(initParams));
+      };
+      createInstance();
+    }
+  }, [connectInstance, initParams]);
```

Comment:
If any of the `initParams` change, a new connect instance will be created. Our recommendation for platforms is to call `update` whenever the `initParams` change, which we aren't doing here. The `initParams` also contain things like the `clientSecret` and `publishableKey`, which are not update-able in a connectjs instance (only locale and appearance are)